pennylane/transforms/adjoint_metric_tensor.py

```diff
@@ -20,41 +20,19 @@
 import pennylane as qml
 from pennylane import numpy as np
 
-# pylint: disable=protected-access
+# pylint: disable=too-many-statements
 from pennylane.transforms.metric_tensor import _contract_metric_tensor_with_cjac
 
 
-def _apply_operations(state, op, device, invert=False):
-    """Wrapper that allows to apply a variety of operations---or groups
-    of operations---to a state or to prepare a new state.
-    If ``invert=True``, this function makes sure not to alter the operations.
-    The state of the device, however may be altered, depending on the
-    device and performed operation(s).
-    """
-    # pylint: disable=protected-access
-    if isinstance(op, (list, np.ndarray)):
-        if invert:
-            op = op[::-1]
-        for _op in op:
-            state = _apply_operations(state, _op, device, invert)
-        return state
-
-    if isinstance(op, qml.QubitStateVector):
-        if invert:
-            raise ValueError("Can't invert state preparation.")
-        device._apply_state_vector(op.parameters[0], op.wires)
-        return device._state
+def _get_op_for_generator(generator, like_real):
+    mat = qml.matrix(generator)
+    real_mat = qml.math.convert_like(mat, like_real)
+    return qml.QubitUnitary(real_mat, generator.wires)
```

Comment:
I got confused while reading this because  `real_mat` is not actually a real matrix.

I think this is due to poor naming (that might be on me, sorry) of `like_real`. While `like_real` is a real-valued object, `convert_like`, which only changes the interface, does not cast to reals.
Maybe, at least within this 4-line function, we could switch to `like_real->like` and `real_mat->mat_interface` or so?

Reply:
Do we actually even need to convert it to the interface? Without this conversion, we could just do:
```
state = apply_operation(generator, state)
```

But I don't understand this history of why we have to convert, so I left it be.

Reply:
Yeah I'm just now using `apply_operation` with the generator`.